Accessions save deletes relations, refs #12662

sbreker opened the pull request:
All relations are being deleted and re-added every time a change to an
accession is saved regardless of which field is updated. This is causing
all related information objects to be reindexed when an accession is
saved - this can be quite slow if many descriptions are linked.

These lines should be comparing the relation "subjectId" and not
"objectId" - objectId contains the accession.id, subjectId contains the
info_obj.id.

sbreker committed:
Accessions save deletes relations, refs #12662

All relations are being deleted and re-added every time a change to an
accession is saved regardless of which field is updated. This is causing
all related information objects to be reindexed when an accession is
saved - this can be quite slow if many descriptions are linked.

These lines should be comparing the relation "subjectId" and not
"objectId" - objectId contains the accession.id, subjectId contains the
info_obj.id.

jraddaoui reviewed (changes requested):
Good catch @sbreker!

I think we don't save any data from related accessions in the ES index for IOs, so we could also set `indexOnSave` to false for [the new relations created after the check](https://github.com/artefactual/atom/blob/dff59ac7eebcd2eb9a62449b4697e17294ea7e2d/plugins/qtAccessionPlugin/modules/accession/actions/editAction.class.php#L330), to avoid a few more unnecessary request to ES.

sbreker committed:
Accessions save deletes relations, refs #12662

Also prevent information object re-indexing when a new information
object is added as an accession reference.

sbreker commented:
Hi @jraddaoui!  As you suggested, I have made the additional change to prevent re-indexing information objects when relations are added to accessions - good idea!

jraddaoui reviewed (approved):
Thanks @sbreker!

jraddaoui commented:
Merged in https://github.com/artefactual/atom/commit/44b295de2883b8821fc5597ac51e3a7ebc9a961d.
